### ISSUE TITLE
Fix `no-descending-specificity` false positives for nested selectors

### ISSUE THREAD

**Issue by bbrk24** [stylelint/stylelint#7723]
### What minimal example or steps are needed to reproduce the bug?

```scss
a {
    color: red;

    .foo & {
        color: blue;
    }

    .foo.bar & {
        color: green;
    }
}

b {
    color: cyan;

    .foo & {
        color: yellow;
    }

    .foo.bar & {
        color: magenta;
    }
}
```

### What minimal configuration is needed to reproduce the bug?

```yml
extends: 
  - stylelint-config-standard-scss
```

### How did you run Stylelint?

`npx stylelint filename.scss` in bash

### Which Stylelint-related dependencies are you using?

```json
    {
        "stylelint": "^16.6.0",
        "stylelint-config-standard-scss": "^13.0.0"
    }
```

### What did you expect to happen?

No errors

### What actually happened?

With the Sass written above, stylelint complains:

> ```
>   16:5  ✖  Expected selector ".foo &" to come before selector ".foo.bar a"  no-descending-specificity
> ```

Obviously, it gives a very similar error message if I put the rules for `b` before the rules for `a`. The only way to avoid this is to manually expand the `&`s and interleave the rules, which defeats the entire point of using Sass.

### Do you have a proposal to fix the bug?

Revert to whatever the logic was in v16.5? There must have been a reason for changing it to what it is now, but with my code I can't get stylelint 16.6 to give me any fewer than *eight* errors for a single Sass file.

**Comment by romainmenke** [comment]
Hi @bbrk24,

Thank you for reporting this and for providing a reproduction.

A slightly more succinct reproduction : https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgOgDMII8AjJAJxQDI0tddIAbCC+FAcwphkwG56OAL5YR2TKTrZ8RCDSkMczVu25QB0sUJAAaEAQCWTGADkkAWziIYADwsAHY3jABnF7vARMhjghAZMFBR0EFscXigXEPYAbSwgoJCXHABPYyYDTBwAWkhvAw5s5KRMKEooEPiUAF1RDzyfADFWcyQcPwArFy8PWHt3RADEkGS0mAys6OCQJjaYZJCdKqTU9MychoKinBKyigqEadnwhZA6oSA

This is most likely an unexpected and unwanted regressions after this change : https://github.com/stylelint/stylelint/pull/7701

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by romainmenke** [comment]
Hi @bbrk24,

We have prepared a patch for this issue and it will be part of the next release in a few days.

For now you can stay a bit longer on 16.5 until the bug fix is publicly available.

Thank you again for reporting this issue.
